modules/master-node-sizing.adoc

```diff
@@ -0,0 +1,39 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/recommended-host-practices.adoc
+
+[id="master-node-sizing_{context}"]
+=  Master node sizing
+
+Control plane density focused tests were run at different node scale to determine the master node resource requirements depending on the number of nodes in the cluster. The recommended sizes are
+
+[options="header",cols="4*"]
+|===
+| Node count |CPU cores |Memory (GB) |AWS Instance Type
+
+| 25
+| 4
+| 16
+| m5.xlarge
+
+| 100
+| 8
+| 32
+| m5.2xlarge
+
+| 250
+| 16
+| 64
+| m5.4xlarge
+
+|===
+
+[NOTE]
+====
+In {product-title} {product-version}, half of a CPU core (500 millicore) is now reserved by the system by default compared to {product-title} 3.11 and previous versions. The sizes are determined taking that into consideration.
+====
+
+[NOTE]
+====
+M5 instances offer better performance with an upgraded CPU, AWS Elastic Network Adapter for faster networking speed and NVMe storage for increased disk reads and write speeds when compared to M4 instances.
```

Comment:
Gone are the days of cloud-provider neutral docs?  I remember I faced a similar issue when documenting [HAProxy performance](https://docs.openshift.com/container-platform/4.1/scalability_and_performance/routing-optimization.html#baseline-router-performance_routing-optimization) on a "typical" cloud provider instance.  We went for something like 4vCPU/16GB RAM in the end.  Are we going to maintain similar docs for GKE/Azure?  Is it possible to keep this a bit more neutral?

Reply:
Thanks for the feedback @jmencak. The idea here is to make it easy for users to know what instance types to use depending on the cloud provider along with resources needed. It's just AWS for now but we need to add instance sizes recommendations for other cloud providers as we move forward.

Thinking this through, it's better to keep it a bit neutral I guess.

Reply:
@vikram-redhat thoughts on this? Should we add cloud provider specific stuff to the doc or just stick with neutral docs?

Reply:
In an ideal world: cloud provider specific. In reality, we may have to stick with neutral docs. 

What we could do as we have a better idea on how popular each cloud provider is with our customer base, we could look at moving to the cloud provider specific docs and adding each provider as the demand rises from the field.